### PR TITLE
Add multi-cluster support

### PR DESCRIPTION
**Pull request by ryansch**
Use case:
- Multiple clusters (for blue/green deploys)
- Multiple regions (in the future)

### CODE REVIEW

**Comment by ryansch** [comment]
Next level client output:

screen shot 2018-01-24 at 10 46 59

File: lib/shipitron/client/fetch_clusters.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      required :cluster_discovery
+
+      def call
+        Resolv::DNS.open do |dns|
+          resources = dns.getresources(
+            context.cluster_discovery,
+            Resolv::DNS::Resource::IN::SRV
+          ).sort_by! do |a|
+            [a.priority, a.weight]
+          end.reverse!
```
Comment (morganick) [inline lib/shipitron/client/fetch_clusters.rb]: Just to make sure I understand what this is doing.

We are going to sort the clusters by priority first and then by weight so for example:
```
[{ priority: 2, weight: 1 }, { priority: 1, weight: 1 }, { priority: 1, weight: 100 }]
```

Then reverse it: 
```
[{ priority: 1, weight: 1 }, { priority: 1, weight: 100 }, { priority: 2, weight: 1 }]
```
I don't know what this data looks like to know whether that's correct or not. Can you provide a sample?

File: lib/shipitron/client/run_ecs_tasks.rb [hunk 2]
```diff
@@ -35,40 +37,55 @@ class RunEcsTasks
...
-                  {
-                    name: 'shipitron',
-                    command: command_args(cluster)
-                  }
-                ]
-              },
-              count: 1,
-              started_by: 'shipitron'
-            )
+        cluster = clusters.first
```
Comment (morganick) [inline lib/shipitron/client/run_ecs_tasks.rb:55]: So we're still only going to be able to ship to one cluster at a time? How do we handle situations when we are shifting weights between two clusters?

File: lib/shipitron/client/run_ecs_tasks.rb [hunk 3]
```diff
@@ -2,6 +2,8 @@
 require 'shipitron/ecs_client'
 require 'shellwords'
 require 'base64'
+require 'tty-table'
+require 'pastel'
```
Comment (morganick) [inline lib/shipitron/client/run_ecs_tasks.rb:6]: That's neat.

File: lib/shipitron/client/fetch_clusters.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+          context.clusters = resources.map do |r|
+            Smash.new(
```
Comment (morganick) [inline lib/shipitron/client/fetch_clusters.rb]: I can't find code to look at for this gem.

File: lib/shipitron/client/fetch_clusters.rb (hunk 1 above)
Comment (ryansch) [reply lib/shipitron/client/fetch_clusters.rb]: screen shot 2018-01-24 at 11 20 42

Comment (ryansch) [reply lib/shipitron/client/fetch_clusters.rb]: https://en.wikipedia.org/wiki/SRV_record

Based on the recommendations there, I sort by lower priority and higher weight in that order.  However, after writing that I'm not sure that I'm sorting correctly.  Needs specs.

File: lib/shipitron/client/run_ecs_tasks.rb (hunk 2 above)
Comment (ryansch) [reply lib/shipitron/client/run_ecs_tasks.rb:55]: No we can ship to multiple clusters but we only need to run the server container once (per region).

File: lib/shipitron/client/fetch_clusters.rb (hunk 4 above)
Comment (ryansch) [reply lib/shipitron/client/fetch_clusters.rb]: It's in shipitron.  See smash.rb.

File: lib/shipitron/client/run_ecs_tasks.rb (hunk 2 above)
Comment (morganick) [reply lib/shipitron/client/run_ecs_tasks.rb:55]: server container == shipitron server container?

File: lib/shipitron/client/fetch_clusters.rb (hunk 4 above)
Comment (ryansch) [reply lib/shipitron/client/fetch_clusters.rb]: Note: newer hashie has support for this as a mixin

Comment (morganick) [reply lib/shipitron/client/fetch_clusters.rb]: Ah ok, cause there is a gem but they don't point to GitHub... so I was like :what:

File: lib/shipitron/client/fetch_clusters.rb (hunk 1 above)
Comment (morganick) [reply lib/shipitron/client/fetch_clusters.rb]: _can now read SRV records_

So in the example above you'd get the following after this code runs:
```
[{ priority: 10, weight: 10 }, { priority: 10, weight: 0 }]
```

File: lib/shipitron/client/run_ecs_tasks.rb [hunk 5]
```diff
@@ -102,14 +119,16 @@ def command_args(cluster)
           '--image-name', escaped(:image_name),
           '--named-tag', escaped(:named_tag),
           '--region', escape(cluster.region),
-          '--cluster-name', escape(cluster.name),
         ].tap do |ary|
+          ary << '--clusters'
+          ary.concat(context.clusters.map {|c| escape(c.name)})
```
Comment (morganick) [inline lib/shipitron/client/run_ecs_tasks.rb]: So this flag here means run on both clusters.

File: lib/shipitron/client/run_ecs_tasks.rb (hunk 2 above)
Comment (ryansch) [reply lib/shipitron/client/run_ecs_tasks.rb:55]: Yeah

File: lib/shipitron/client/run_ecs_tasks.rb (hunk 5 above)
Comment (ryansch) [reply lib/shipitron/client/run_ecs_tasks.rb]: It means update services for both clusters.

**Comment by ryansch** [comment]
@morganick Once more with feeling.  I think this is ready.

**Review by morganick — APPROVED** [review]

**Pull request merged**